x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -395,21 +359,20 @@ public void parse(ParseContext context) throws IOException {
                     "[" + COUNTS_FIELD.getPreferredName() +"] but got [" + values.size() + " != " + counts.size() +"]");
             }
             if (fieldType().hasDocValues()) {
-                BytesStreamOutput streamOutput = new BytesStreamOutput();
+                ByteBuffersDataOutput dataOutput = new ByteBuffersDataOutput();
                 for (int i = 0; i < values.size(); i++) {
                     int count = counts.get(i);
                     if (count < 0) {
                         throw new MapperParsingException("error parsing field ["
                             + name() + "], ["+ COUNTS_FIELD + "] elements must be >= 0 but got " + counts.get(i));
                     } else if (count > 0) {
                         // we do not add elements with count == 0
-                        streamOutput.writeVInt(count);
-                        streamOutput.writeDouble(values.get(i));
+                        dataOutput.writeVInt(count);
+                        dataOutput.writeLong(NumericUtils.doubleToSortableLong(values.get(i)));
```

Comment:
Since we don't need ordering, let's just do `Double.doubleToRawLongBits`, which is cheaper.